client/lib/upgrades/actions/domain-management.js

```diff
@@ -126,17 +125,17 @@ function fetchDomains( siteId ) {
 		} );
 	}
 
+	const domains = DomainsStore.getBySite( siteId );
+	if ( domains.isFetching || domains.hasLoadedFromServer ) {
```

Comment:
@rads Do you remember why the `hasLoadedFromServer` was used in fetchDns/fetchNameservers? Does it make sense to use it for `DomainsStore` also?